python/modem-firmware-1.3+/rtt_interface.py

```diff
@@ -0,0 +1,138 @@
+import time
+from pynrfjprog import LowLevel
+
+CRLF = '\r\n'
+
+def send_rtt(api, tx_data):
+    for i in range(0, len(tx_data), 12):
+       api.rtt_write(channel_index=0, msg=tx_data[i:i+12])
+       time.sleep(0.01)
+
+def readlines_at_rtt(api, timeout_s):
+    elapsed_s = 0
+    rx = ''
+
+    while elapsed_s < timeout_s:
+        rx_new = api.rtt_read(channel_index=0, length=4096)
+        if rx_new:
+            rx = rx + rx_new
+
+            if rx:
+                last_line = rx.splitlines().pop()
+                if last_line == 'OK' or last_line == 'ERROR':
+                    break
+
+        time.sleep(0.1)
+        elapsed_s = elapsed_s + 0.1
+
+    if elapsed_s >= timeout_s:
+        print('RTT read timeout')
+
+    return rx.splitlines(keepends=True)
```

Comment:
Behavior might  be more symmetric with the serial interface if you implement readline rather than readlines.

Then again, if you implement serial_interface.py, you could go in the reverse direction and use readlines for both

Personally I think I prefer the readline approach

Reply:
i'll add this to the new ticket